xds/internal/balancer/clusterimpl/tests/balancer_test.go

```diff
@@ -170,3 +178,207 @@ func (s) TestConfigUpdateWithSameLoadReportingServerConfig(t *testing.T) {
 		t.Fatal("New LRS stream created when expected not to")
 	}
 }
+
+// TestLoadReportingPickFirstMultiLocality tests whether load is reported correctly
+// when using pickfirst with endpoints in multiple localities.
+func (s) TestLoadReportingPickFirstMultiLocality(t *testing.T) {
+	originalTickerFactory := transport.TickerFactory
+	tickChan := make(chan time.Time)
+	transport.TickerFactory = func(freq time.Duration) (<-chan time.Time, func()) {
+		return tickChan, func() {}
+	}
+	defer func() {
+		transport.TickerFactory = originalTickerFactory
+	}()
+
+	// Create an xDS management server that serves ADS and LRS requests.
+	mgmtServer := e2e.StartManagementServer(t, e2e.ManagementServerOptions{SupportLoadReportingService: true})
+
+	// Create bootstrap configuration pointing to the above management server.
+	nodeID := uuid.New().String()
+	bc := e2e.DefaultBootstrapContents(t, nodeID, mgmtServer.Address)
+	testutils.CreateBootstrapFileForTesting(t, bc)
+
+	// Create an xDS resolver with the above bootstrap configuration.
+	var resolverBuilder resolver.Builder
+	var err error
+	if newResolver := internal.NewXDSResolverWithConfigForTesting; newResolver != nil {
+		resolverBuilder, err = newResolver.(func([]byte) (resolver.Builder, error))(bc)
+		if err != nil {
+			t.Fatalf("Failed to create xDS resolver for testing: %v", err)
+		}
+	}
+
+	// Start two server backends exposing the test service.
+	server1 := stubserver.StartTestService(t, nil)
+	defer server1.Stop()
+
+	server2 := stubserver.StartTestService(t, nil)
+	defer server2.Stop()
+
+	port1 := testutils.ParsePort(t, server1.Address)
+	port2 := testutils.ParsePort(t, server2.Address)
+
+	// Configure the xDS management server with default resources. Override the
+	// default cluster to include an LRS server config pointing to self.
+	const serviceName = "my-test-xds-service"
+	routeConfigName := "route-" + serviceName
+	clusterName := "cluster-" + serviceName
+	endpointsName := "endpoints-" + serviceName
+	resources := e2e.UpdateOptions{
+		NodeID:    nodeID,
+		Listeners: []*v3listenerpb.Listener{e2e.DefaultClientListener(serviceName, routeConfigName)},
+		Routes:    []*v3routepb.RouteConfiguration{e2e.DefaultRouteConfig(routeConfigName, serviceName, clusterName)},
+		Clusters: []*v3clusterpb.Cluster{
```

Comment:
Also mention in the same comment that the cluster resource is defined with a custom load balancing policy set to `pickfirst`.

Reply:
Done.